### PR TITLE
[ROCM] Avoid hard coded specifications for mi300 transpose fusion

### PR DESCRIPTION
**Pull request by Ruturaj4**

### CODE REVIEW

**Review by akuegel — APPROVED** [review]

**Review by akuegel — APPROVED** [review]
Just one question: the PR description says we can avoid some hardcoded specifications, but it seems they are not removed in this PR? Is this planned in a followup PR, or do I misunderstand something?

File: xla/service/gpu/fusions/transpose.cc [hunk 1]
```diff
@@ -77,6 +78,18 @@ Tiling ComputeTransposeTiling(const TransposeDescription& tiled_transpose) {
   absl::InlinedVector<int64_t, 4> num_threads{1, 1, WarpSize()};
```
Comment (i-chaochen) [inline xla/service/gpu/fusions/transpose.cc:78]: since `gpu_device_info.threads_per_warp()` is already there, could you replace this `WarpSize()` as well?

**Comment by i-chaochen** [comment]
Thanks for the review! Yes, there are more changes on tiling code as well.

**Comment by akuegel** [comment]
> Thanks for the review! Yes, there are more changes on tiling code as well.

Ok, I think I understand what was meant: before, it did always choose one specific tiling config. Now, it chooses a different one for mi300, but it still chooses the other hardcoded values in other cases. Somehow I thought i would see some hardcoded values being removed, but I guess we still need it in other cases.

**Comment by i-chaochen** [comment]
Oh I guess you're refering this one? https://github.com/openxla/xla/pull/12494

**Pull request closed without merging**